src/block_disagg/block_disagg_read.c

```diff
@@ -167,9 +167,7 @@ __block_disagg_read_multiple(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *block_di
         if (swap.checksum == checksum) {
             blk->checksum = 0;
             if (__wt_checksum_match(current->data,
-                  F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP, checksum) &&
-              swap.reconciliation_id == reconciliation_id) {
```

Comment:
@ddanderson I think I fixed the checksum verification bug. Can you check if this makes sense?

Reply:
swap.reconciliation_id == reconciliation_id is always false except for the last data.

Reply:
Ah, yes - the reconciliation equality check should only be done for the "final piece" (he last delta" or full page if there is no delta).  Does this alternate change work for your tests?
```
      if (__wt_checksum_match(current->data,
           F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP, checksum) &&
           (result != last || swap.reconciliation_id == reconciliation_id)) {
```
Reason: In the code several lines above this, we see if `swap.reconciliation_id < reconciliation_id ` and do a retry if so (only done for the final piece).  However, it's still possible that `swap.reconciliation_id > reconciliation_id` which would be an error.  But it only makes sense to do that check on the final piece.

Reply:
test_oligarch09 now runs but test_oligarch08 still fail. I think it is a different problem.

Reply:
Okay for now, I'll want to investigate after M2.